internal/runtime/client/client.go

```diff
@@ -154,10 +154,14 @@ func (c *client) Discover(ctx context.Context, extensionConfig *runtimev1.Extens
 	modifiedExtensionConfig.Status.Handlers = []runtimev1.ExtensionHandler{}
 
 	for _, handler := range response.Handlers {
+		handlerName, err := NameForHandler(handler, extensionConfig)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to discover extension")
```

Comment:
```suggestion
			return nil, errors.Wrapf(err, "failed to discover extension %q", extensionConfig.Name
```

(for consistency with above)